### DISCUSSION TITLE
Why does Nanobind restrict Stable ABI usage to 3.12+?

### DISCUSSION THREAD

**Discussion by Ahajha** [wjakob/nanobind#826] [Q&A] [answered]
I'm wondering why nanobind restricts use of the stable ABI to 3.12+. My understanding is that in theory it doesn't need this restriction, but I imagine there's something important to 3.12 that enabled something necessary. Ideally I would like to ship a single binary for 3.9+.

**Comment by wjakob** [comment, marked as answer]
Link to previous discussion: https://github.com/wjakob/nanobind/discussions/642
